### DISCUSSION TITLE
Rules for `related_road_events` of work zone detours

### DISCUSSION THREAD

**Discussion by sagarrat7** [usdot-jpo-ode/jpo-wzdx#368] [Q&A]
Are there rules to follow when relating road events to each other? In [Scenario 4](https://raw.githubusercontent.com/usdot-jpo-ode/wzdx/main/examples/WorkZoneFeed/scenario4_detour_linestring_example.geojson), the work zone event (id `a15f7570-b7e6-4367-8ad9-3a462eea65dd`) is related to the detour event with id `cf1092ba-3b8d-4e91-81ef-daa4a98662e1`.

Why this segment, and not any of the other two detour segments? 

That detour segment (id `cf1092ba-3b8d-4e91-81ef-daa4a98662e1`) is related to both the work zone event and another detour, `4d151e7d-11d8-4b99-a192-51e189da0de7`, which is marked as next-in-sequence. Why is this sequence? 

This detour segment (id `4d151e7d-11d8-4b99-a192-51e189da0de7`) has these detour relations:

```
                 {
                     "type": "first-in-sequence",
                     "id": "cf1092ba-3b8d-4e91-81ef-daa4a98662e1"
                  },
                  {
                     "type": "next-in-sequence",
                     "id": "9436226a-01b0-47ff-8a13-670e87549458"
                  }
```

Why is one of these first vs next? Is it related to the flow of traffic, or is it arbitrary?

**Comment by mark-mockett** [comment]
@sagarrat7 thanks for the question - there are a few things to unpack here:

**first vs. next:** The `first-in-sequence` and `next-in-sequence` relation types are used to identify road events that are grouped together linearly in space (or in time, if events are continuous; recurrences of a work zone with breaks in between should use the `first-occurrence` and `next-occurrence` types of relation). The `first-in-sequence` relationship be used to refer back to the first event in any sequence so that events can be easily identified as a group.

In this example, the sequence of detour events capture the full detour route as a driver experiences it. To drive around the work zone on I-35, a driver will use road F22, then US 69, then IA 210. Each of those segments is represented as a different detour road event. The first event (detour segment on F22), will be referred to as such by the other two segments. Each segment points to the next segment in the detour (F22 -> US 69, US 69 -> IA 210).

![image](https://github.com/usdot-jpo-ode/wzdx/assets/59966277/3702d304-5414-42c5-9018-ec6747d7ac1a)

**relating with other work zones/detours:** There's some guidance in the header to the [RelatedRoadEventType](https://github.com/usdot-jpo-ode/wzdx/blob/main/spec-content/enumerated-types/RelatedRoadEventType.md) - if a work zone or detour event is part of a sequence, we recommend only pointing to the first in a sequence. The first road event in a sequence or recurrence is treated as the "head" of a group and pointing only to the first event minimizes the number of relationships specified. In these situations, end users will need to find the "first" road event and iterate through all linked "next" road events to find all related road events.

**overall:** None of these are "rules", per se - you could use even fewer relationships to get the same structure across, but this approach balances the amount of information shared with the manipulation required to define all the relationships. Here's a color-coded diagram of how all of these relationships are used in this example: 

![image](https://github.com/usdot-jpo-ode/wzdx/assets/59966277/c035b658-a7e2-45eb-ae85-cbbb2f16c1be)
